---
date: "2018-11-05T00:00:00+01:00"
draft: false
linktitle: No. 1 Tuesday
menu:
  bocas-del-toro-panama-2019:
    parent: Topics
    weight: 1
title: Defining & Sampling a Microbiome
toc: true
type: docs
weight: 1
---

## Overview

In the literature, core microbiomes are broadly defined as assemblages of microbial taxa that are consistently found across individuals of the same species. They form intimate and specialized relationships with their host.

## General Discussion

What are the attributes of a core microbiome? Do all metazoan have a core microbiomes? Are some microbial taxa consistently found across hosts of a community? And are they functionally important or opportunistic taxa?

How to characterize the core microbiome of all hosts within a community (i.e., methodological considerations for comparative approaches)? How do marine hosts acquire this core community? How to account for environmental (oceanographic) factors impacting this core or masking the detection of the core community?

## Discussion Questions

#### 1A: What is a core microbiome?

Group members, moderator, note taker/presenter

#### 1B: How to characterize the core microbiome?

Group members, moderator, note taker/presenter

## References

## Outcomes

**1A: What is a core microbiome?**

Groups 1 and 2

1. What is a microbiome? All microbial organisms (e.g. bacteria, archaea, viruses, fungi,
protozoans, and small microbial eukaryotes) “somehow” related/associated with an
organism/host or found in a particular habitat/ecosystem in a given place and time.


2. Is there a need to define/separate what is environmental microbiome, that is, the
microbial community structure in a habitat (e.g. soil microbiome, water column
microbiome) from host microbiome, that is, the microbial community structure
associated to an organism (e.g. coral microbiome).

3. What is the core microbiome? Microbes constantly (i.e. highly frequent and
abundant/dominant) found in a habitat/host. In this sense, stability (i.e. taxon
composition) and functionality of the microbial community is maintained within a
defined scale.

4. Is symbiosis between microbes and host a “must” to be considered as core
microbiome? What about the environment? Is there a need to differentiate/define
what is the “core taxonomic” (i.e. consistent microbial) members and “core
functional” (i.e. genes or pathways) microbiome? Super organisms to neutral
models, context dependent interactions.

5. How can we tease a part and potentially quantify the direct and indirect effects of
certain microbial groups/genes within the microbiome?

6. Should use a metric(s) to define the core microbiome? Relative abundance (> a
specific cut-off, e.g. 80%), spatial and temporal frequency (found in most samples or
all samples), or simply presence/absence.

7. Large microbiome picture: Community similarity analysis: gain and loss of
taxonomic/functional core microbiome across different hosts through time. At what
level the microbiome is shared in a given specific group (i.e. at the tips of the host
phylogeny; species vs. higher taxonomic ranks). In addition, network analysis across
different groups to characterize a particular habitat/system.

8. What are some of the pros and cons associated to the concepts of microbiome and
core microbiome?

**1B: How to characterize the core microbiome?**

Groups 3 and 4

1. How to characterize the core microbiome? In addition to the considerations on how we define the core microbiome, spatio-temporal variation is a key aspect when characterizing the core microbiome. The spatial component might give us an idea of “how common/spread ” such microbial taxa and associations are in a particular host/habitat, whereas the temporal component might inform us on how stable/frequent these associations are.

2. Long-term studies are crucial to identify community patterns and also to detect drastic shifts.

3. Baseline studies for different habitats/hosts and archiving material for future analyses can also advance our understanding. In addition, these baseline studies might direct us to more specific questions/goals.

4. Taxonomic scale of both host and microbial taxa should be considered, potentially tested using different resolution levels. What taxonomic resolution should we be considering when characterizing microbiomes?

5. Organism and ecosystem health should be evaluated, particularly the links between microbiome and their function

6. Stability and variability (both in space and time) of the microbiome should be explored

7. Mapping the landscape of the host/ecosystem (e.g. strata, body regions, organism geography/distribution), life stage, life history and other important aspects of the system. This would allow us to fully/properly characterize the microbiome. Additionally, a comparison between whole/integrated vs. partitioned microbiomes or microbiome/tissue specificity can be determined.

8. How important is to characterize the background environmental (i.e. sediment, water, etc.) to understand the core microbiome of an organism? How do they acquire/select their microbiome?

9. How to standardize sampling strategies for microbiome studies? The importance of standard protocols, particularly for worldwide scale studies (e.g. ZEN network).

10. What is the importance of well-curate databases for microbiome studies? 16S, 18S, ITS datasets (who is there)? Who is there and what they potentially do?

11. What sort of datasets should we be producing (e.g. metabarcoding/amplicon, transcriptomic, and metagenomics)? What are the other techniques we might consider to use (e.g. microscopy)?


**Random Comments:**

One comment I wanted to provide regarding core microbiome and healthy microbiome is that I think studies often don't examine enough individuals to really address this question.  Sample sets of hundreds of individuals provide much clearer patterns compared to only a handful of individuals.  In corals and other highly permitted animals, it is often difficult to get access to these larger number of individuals, and we need managers and permitters to understand how necessary this is, especially today as ocean conditions (and likely microbiomes) will only change more in the future.

Another general comment...I also see a trend in the journals and at meetings in that descriptive studies (such as core microbiome descriptions) are now often viewed as less important than those that provide more in-depth functional trends in symbiotic interactions.  Yet, we still need descriptive studies in marine animal microbiome research!  And these types of studies are perfect for training undergraduates and graduate students and provide context for other researchers to establish new projects upon.  I am not sure how to change the culture, but perhaps this is a point that could be make in the Outcome article that descriptive studies are still incredibly valuable.

## Schedule

#### 0700--0830 Breakfast


#### 0900--1200 Topic 1

Jarrod will get Becky from the airport

#####  0900--0915

Introduction to Topic 1 (Jarrod)

#####  0915--1015

##### <font color="red">**Break-out groups**</font>

* 1A: **What is a core microbiome?**
* 1B: **How to characterize the core microbiome?**

#### 1015--1045 Break

##### 1045--1115

Report and discuss ideas from topics


##### 1115--1200

Summary and discussion of morning sessions

#### 1200--1630 Field trip

 with lunch in the field

##### 1630--1730

Debrief

#### 1800--1930 Dinner


##### 1930--2100

Evening Discussion

* Discuss sampling strategies from in the field.
* How to standardize it. Storage and extraction protocols.
* Do common protocols introduce bias and selection?
* Which species are missed?
* Would it make sense to produce guidelines/website/protocols?
