---
title: Test-Operational
linktitle: Test-Operational
summary:
date: 2019-12-13T15:58:49-05:00
lastmod: 2019-12-13T15:58:49-05:00
draft: true  # Is this a draft? true/false
toc: true  # Show table of contents? true/false
type: docs  # Do not modify.
weight: 5
raw: "https://raw.githubusercontent.com/rbind/blogdown-demo/master/content/post/2017-09-07-adding-htmlwidgets-to-r-markdown-posts.Rmd"

# Add menu entry to sidebar.
# - Substitute `example` with the name of your course/documentation folder.
# - name: Declare this menu item as a parent with ID `name`.
# - parent: Reference a parent ID if this page is a child.
# - weight: Position of link in menu.
bibliography: [cite.bib]
url_pdf: http://arxiv.org/pdf/1512.04133v1
link-citations: true
menu:
  trans-water:
    parent: Environment Setup
    # parent: YourParentID
    weight: 5
---

```{r, setup, echo=FALSE}
# For DAG
library(DiagrammeR)
library(htmlwidgets)
library(DiagrammeRsvg)
library(rsvg)

```

```{r, workflow, echo = FALSE}
dag <- grViz ("
digraph boxes_and_circles {
  graph [layout = dot]

  node [shape = rectangle, style = rounded, fontname=sans, fontsize=12, penwidth=4]
  edge[penwidth=4, color=grey];

0[label = 'metagenomics_workflow_target_rule', color = 'grey', style='box'];
1[label = 'anvi_merge', color = '#CC79A7', style='filled, rounded'];
2[label = 'anvi_merge', color = '#CC79A7', style='filled, rounded'];
3[label = 'annotate_contigs_database', color = '#E69F00', style='rounded'];
4[label = 'annotate_contigs_database', color = '#E69F00', style='rounded'];
5[label = 'gen_qc_report', color = '#56B4E9', style='rounded'];
6[label = 'anvi_cluster_contigs\ndriver: concoct', color = '#F0E442', style='rounded'];
7[label = 'anvi_cluster_contigs\ndriver: concoct', color = '#F0E442', style='rounded'];
8[label = 'anvi_gen_contigs_database', color = '#CC79A7', style='filled, rounded'];
9[label = 'anvi_profile', color = '#CC79A7', style='filled, rounded'];
10[label = 'anvi_profile', color = '#CC79A7', style='filled, rounded'];
11[label = 'import_percent_of_reads_mapped', color = '#009E73', style='rounded'];
12[label = 'import_percent_of_reads_mapped', color = '#009E73', style='rounded'];
13[label = 'import_krakenuniq_taxonomy', color = '#E69F00', style='rounded'];
14[label = 'import_krakenuniq_taxonomy', color = '#E69F00', style='rounded'];
15[label = 'anvi_gen_contigs_database', color = '#CC79A7', style='filled, rounded'];
16[label = 'anvi_profile', color = '#CC79A7', style='filled, rounded'];
17[label = 'anvi_profile', color = '#CC79A7', style='filled, rounded'];
18[label = 'import_percent_of_reads_mapped', color = '#009E73', style='rounded'];
19[label = 'import_percent_of_reads_mapped', color = '#009E73', style='rounded'];
20[label = 'import_krakenuniq_taxonomy', color = '#E69F00', style='rounded'];
21[label = 'import_krakenuniq_taxonomy', color = '#E69F00', style='rounded'];
22[label = 'anvi_import_taxonomy_for_genes', color = '#E69F00', style='rounded'];
23[label = 'anvi_run_hmms', color = '#E69F00', style='rounded'];
24[label = 'anvi_run_ncbi_cogs', color = '#E69F00', style='rounded'];
25[label = 'anvi_run_scg_taxonomy', color = '#E69F00', style='rounded'];
26[label = 'anvi_run_pfams', color = '#E69F00', style='rounded'];
27[label = 'anvi_import_taxonomy_for_genes', color = '#E69F00', style='rounded'];
28[label = 'anvi_run_hmms', color = '#E69F00', style='rounded'];
29[label = 'anvi_run_ncbi_cogs', color = '#E69F00', style='rounded'];
30[label = 'anvi_run_scg_taxonomy', color = '#E69F00', style='rounded'];
31[label = 'anvi_run_pfams', color = '#E69F00', style='rounded'];
32[label = 'iu_filter_quality_minoche\nsample: EPR_9A', color = '#56B4E9', style='rounded'];
33[label = 'iu_filter_quality_minoche\nsample: EPR_11A', color = '#56B4E9', style='rounded'];
34[label = 'iu_filter_quality_minoche\nsample: WAR_MAR', color = '#56B4E9', style='rounded'];
35[label = 'iu_filter_quality_minoche\nsample: WAM_MYS', color = '#56B4E9', style='rounded'];
36[label = 'anvi_script_reformat_fasta', color = '#56B4E9', style='rounded'];
37[label = 'anvi_init_bam', color = '#009E73', style='rounded'];
38[label = 'anvi_init_bam', color = '#009E73', style='rounded'];
39[label = 'krakenuniq_mpa_report', color = '#E69F00', style='rounded'];
40[label = 'krakenuniq_mpa_report', color = '#E69F00', style='rounded'];
41[label = 'anvi_script_reformat_fasta', color = '#56B4E9', style='rounded'];
42[label = 'anvi_init_bam', color = '#009E73', style='rounded'];
43[label = 'anvi_init_bam', color = '#009E73', style='rounded'];
44[label = 'krakenuniq_mpa_report', color = '#E69F00', style='rounded'];
45[label = 'krakenuniq_mpa_report', color = '#E69F00', style='rounded'];
46[label = 'centrifuge', color = '#E69F00', style='rounded'];
47[label = 'centrifuge', color = '#E69F00', style='rounded'];
48[label = 'iu_gen_configs', color = '#56B4E9', style='rounded'];
49[label = 'anvi_script_reformat_fasta_prefix_only', color = '#56B4E9', style='rounded'];
50[label = 'samtools_view', color = '#009E73', style='rounded'];
51[label = 'samtools_view', color = '#009E73', style='rounded'];
52[label = 'krakenuniq', color = '#E69F00', style='rounded'];
53[label = 'krakenuniq', color = '#E69F00', style='rounded'];
54[label = 'anvi_script_reformat_fasta_prefix_only', color = '#56B4E9', style='rounded'];
55[label = 'samtools_view', color = '#009E73', style='rounded'];
56[label = 'samtools_view', color = '#009E73', style='rounded'];
57[label = 'krakenuniq', color = '#E69F00', style='rounded'];
58[label = 'krakenuniq', color = '#E69F00', style='rounded'];
59[label = 'export_gene_calls_for_centrifuge', color = '#E69F00', style='rounded'];
60[label = 'export_gene_calls_for_centrifuge', color = '#E69F00', style='rounded'];
61[label = 'megahit\ngroup: EP', color = '#56B4E9', style='rounded'];
62[label = 'bowtie', color = '#009E73', style='rounded'];
63[label = 'bowtie', color = '#009E73', style='rounded'];
64[label = 'gzip_fastqs\nR: R1', color = '#56B4E9', style='rounded'];
65[label = 'gzip_fastqs\nR: R2', color = '#56B4E9', style='rounded'];
66[label = 'gzip_fastqs\nR: R1', color = '#56B4E9', style='rounded'];
67[label = 'gzip_fastqs\nR: R2', color = '#56B4E9', style='rounded'];
68[label = 'megahit\ngroup: WA', color = '#56B4E9', style='rounded'];
69[label = 'bowtie', color = '#009E73', style='rounded'];
70[label = 'bowtie', color = '#009E73', style='rounded'];
71[label = 'gzip_fastqs\nR: R1', color = '#56B4E9', style='rounded'];
72[label = 'gzip_fastqs\nR: R2', color = '#56B4E9', style='rounded'];
73[label = 'gzip_fastqs\nR: R1', color = '#56B4E9', style='rounded'];
74[label = 'gzip_fastqs\nR: R2', color = '#56B4E9', style='rounded'];
75[label = 'bowtie_build', color = '#009E73', style='rounded'];
76[label = 'bowtie_build', color = '#009E73', style='rounded'];

1->0; 2->0; 3->0; 4->0; 5->0; 6->0; 7->0; 8->1; 9->1;
10->1; 11->1; 12->1; 13->1; 14->1; 15->2; 16->2; 17->2;
18->2; 19->2; 20->2; 21->2; 8->3; 22->3; 23->3; 24->3;
25->3; 26->3; 15->4; 27->4; 28->4; 29->4; 30->4; 31->4;
32->5; 33->5; 34->5; 35->5; 8->6; 1->6; 15->7; 2->7;
36->8; 37->9; 8->9; 38->10; 8->10; 9->11; 10->12; 39->13;
11->13; 9->13; 40->14; 12->14; 10->14; 41->15; 42->16; 15->16;
43->17; 15->17; 16->18; 17->19; 44->20; 18->20; 16->20; 45->21;
19->21; 17->21; 46->22; 8->22; 8->23; 8->24; 23->25; 8->25;
8->26; 47->27; 15->27; 15->28; 15->29; 28->30; 15->30; 15->31;
48->32; 48->33; 48->34; 48->35; 49->36; 50->37; 51->38; 52->39;
53->40; 54->41; 55->42; 56->43; 57->44; 58->45; 59->46; 60->47;
61->49; 62->50; 63->51; 64->52; 65->52; 66->53; 67->53; 68->54;
69->55; 70->56; 71->57; 72->57; 73->58; 74->58; 8->59; 15->60;
64->61; 66->61; 65->61; 67->61; 75->62; 64->62; 65->62; 75->63;
66->63; 67->63; 32->64; 32->65; 33->66; 33->67; 71->68; 73->68;
72->68; 74->68; 76->69; 71->69; 72->69; 76->70; 73->70; 74->70;
34->71; 34->72; 35->73; 35->74; 36->75; 41->76

	graph [nodesep = 0.1]

}
")

export_svg(dag) %>%
  charToRaw() %>%
  rsvg() %>%
  png::writePNG("../../../static/img/dag.png")

```

`r blogdown::shortcode("figure", src = "/img/dag.png", title = " Colors indicate broad divisions of workflow: sky blue, short-read prep & co-assembly; blueish green, short-read mapping to assembly; orange, taxonomic or functional classification; yellow, automatic binning. Filled reddish purple boxes indicate steps where databases are constructed. ", lightbox = "true", alt = "Hum, looks like there is something wrong here.")`

Directed acyclic graph (DAG) of the metagenomic workflow where edge connections represent dependencies and nodes represent commands. The workflow begins with raw data   (trimmed of Illumina adapters) and continues up to and including automatic binning of contigs. At the end of this workflow we then proceed with manual binning and MAG generation. For simplicity, only 2 samples from the Eastern Pacific & 2 samples from the Western Atlantic are shown.

citation [@eren2015anvi]
