---
title: ProjectDigest
subtitle: Intestinal microbes as an axis of functional diversity among large marine consumers
authors: []
summary: Amplicon analysis of intenstinal microbiomes from five species of herbivorous fish of a Florida reef.
tags: ["fish", "16S rRNA", "amplicon"]
categories: []
date: "2019-12-05T00:00:00+01:00"
external_link:
bibliography: [cite.bib]
header:
  caption: ""
  image: "banners/florida_fish.png"
image:
  caption: ""
  focal_point:
  preview_only: true
links:
- name: Go to Project Site
  url: https://projectdigest.github.io/
- icon: database
  icon_pack: fas
  name: Data
  url: https://projectdigest.github.io/data_availability.html
- icon: code
  icon_pack: fas
  name: Code
  url: https://projectdigest.github.io/raw_code.txt
- icon: github
  icon_pack: fab
  name: GitHub
  url: https://github.com/projectdigest/web/
- icon: newspaper
  icon_pack: fas
  name: Publication
  url: https://projectdigest.github.io/publication.html
publication: null
slides:
url_code: ""
url_pdf: ""
url_slides: ""
url_video: ""

weight: 1
---

Welcome. This page provides a link to the <em>external</em> website for our study on the intestinal microbiome of herbivorous reef fish entitled <em>Intestinal microbes: as axis of functional diversity among large marine consumers</em>. Microbes are ubiquitous throughout the worldâ€™s ocean yet the extent of their influence on the ecology and evolution of large, mobile fauna is poorly understood.
<br/>

To access the complete bioinformatic workflow & raw data use the buttons at the top of the page or visit
>https://projectdigest.github.io/

{{% alert synopsis %}}

SYNOPSIS
<hr>
In this study, we

- Conducted field observations, which showed that five common Caribbean fish species display marked differences in where they feed and what they feed on.
- Used 16S rRNA amplicon analysis of intestinal microbiomes from 50 individual fish across the 5 species to show a high degree of species-specific composition.
- Performed phylogenetic comparison of  dominant microbiome members to show that intestinal microbiome are a mixture of  environmental generalists, animal-associates, and fish-specialists.
- Found that the closest relatives of fish-specialists were microbes from intestines of ecologically and evolutionarily related fishes in other, distant oceans.

{{% /alert %}}

On this site you will find details on our reproducible bioinformatics workflow and how to access all data/data products. We also provide numerous data tables and figures including all tables, figures, and supplementary information from the original publiation.

Use the `Cite` button above to access a BibTeX formatted bibliography for the references we used in this study.

<hr/>

<p><small>
<em>All banner photos on this page were retreived from Wikimedia Commons and Licensed under Creative Commons Attributions. Each photo was modified from the originals by removing the background</em>.

**Banner photos from left to right**: <em>Acanthurus bahianus/Acanthurus tractus </em><a href="https://commons.wikimedia.org/wiki/Category:Acanthurus_bahianus#/media/File:Acanthurus_bahianus_SI2.jpg">(original)</a> licenced under <a href="https://creativecommons.org/share-your-work/public-domain/cc0">CC-0</a>; <em>Acanthurus coeruleus </em><a href="https://commons.wikimedia.org/wiki/File:Fish13092009a.jpg">(original)</a> by Sylfred1977 licenced under <a href="https://creativecommons.org/licenses/by-sa/3.0/">CC-BY-SA-3.0</a>; <em>Scarus taeniopterus</em> (Desmarest 1831) <a href="https://commons.wikimedia.org/wiki/File:Scarus_taeniopterus_-_pone.0010676.g130.png">(original)</a>, <em>Sparisoma aurofrenatum</em> (Valenciennes, 1840) <a href="https://commons.wikimedia.org/wiki/File:Sparisoma_aurofrenatum_-_pone.0010676.g133.png">(original)</a>, and <em>Sparisoma viride</em> (Bonnaterre, 1788) <a href="https://commons.wikimedia.org/wiki/File:Stoplight-parrotfish.jpg">(original)</a> all licenced under <a href="https://creativecommons.org/licenses/by/2.5/">CC-BY-2.5</a> by <a href="https://doi.org/10.1371/journal.pone.0010676">Williams et. al., 2010</a>.
</small></p>
